src/main/java/com/arpnetworking/metrics/incubator/impl/TsdPeriodicMetricsFactory.java

```diff
@@ -0,0 +1,251 @@
+        NonClosingMetrics(final Metrics wrapped) {
+            this._wrapped = wrapped;
+        }
+
+        @Override
+        public Counter createCounter(@Nonnull final String name) {
+            return _wrapped.createCounter(name);
+        }
+
+        @Override
+        public void incrementCounter(@Nonnull final String name) {
+            _wrapped.incrementCounter(name);
+        }
+
+        @Override
+        public void incrementCounter(@Nonnull final String name, final long value) {
+            _wrapped.incrementCounter(name, value);
+        }
+
+        @Override
+        public void decrementCounter(@Nonnull final String name) {
+            _wrapped.decrementCounter(name);
+        }
+
+        @Override
+        public void decrementCounter(@Nonnull final String name, final long value) {
+            _wrapped.decrementCounter(name, value);
+        }
+
+        @Override
+        public void resetCounter(@Nonnull final String name) {
+            _wrapped.resetCounter(name);
+        }
+
+        @Override
+        public Timer createTimer(@Nonnull final String name) {
+            return _wrapped.createTimer(name);
+        }
+
+        @Override
+        public void startTimer(@Nonnull final String name) {
+            _wrapped.startTimer(name);
+        }
+
+        @Override
+        public void stopTimer(@Nonnull final String name) {
+            _wrapped.stopTimer(name);
+        }
+
+        @Override
+        public void setTimer(@Nonnull final String name, final long duration, @Nullable final TimeUnit unit) {
+            _wrapped.setTimer(name, duration, unit);
+        }
+
+        @Override
+        public void setTimer(@Nonnull final String name, final long duration, @Nullable final Unit unit) {
+            _wrapped.setTimer(name, duration, unit);
+        }
+
+        @Override
+        public void setGauge(@Nonnull final String name, final double value) {
+            _wrapped.setGauge(name, value);
+        }
+
+        @Override
+        public void setGauge(@Nonnull final String name, final double value, @Nullable final Unit unit) {
+            _wrapped.setGauge(name, value, unit);
+        }
+
+        @Override
+        public void setGauge(@Nonnull final String name, final long value) {
+            _wrapped.setGauge(name, value);
+        }
+
+        @Override
+        public void setGauge(@Nonnull final String name, final long value, @Nullable final Unit unit) {
+            _wrapped.setGauge(name, value, unit);
+        }
+
+        @Override
+        public void addAnnotation(@Nonnull final String key, @Nonnull final String value) {
+            _wrapped.addAnnotation(key, value);
+        }
+
+        @Override
+        public void addAnnotations(@Nonnull final Map<String, String> map) {
+            _wrapped.addAnnotations(map);
+        }
+
+        @Override
+        public boolean isOpen() {
+            return _wrapped.isOpen();
+        }
+
+        @Override
+        public void close() {
```

Comment:
So I think one differentiator to the current design is that I think we need to extract the common metrics interface between periodic and event based, and then apply additional methods to each as necessary. It seems you believe the only difference is that event based can be closed. How did you resolve the issue with counter instances that span collection periods for the closable variant?